codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/DocumentClientCommandGenerator.java

```diff
@@ -0,0 +1,380 @@
+    private final Model model;
+    private final OperationShape operation;
+    private final SymbolProvider symbolProvider;
+    private final TypeScriptWriter writer;
+    private final Symbol symbol;
+    private final OperationIndex operationIndex;
+    private final String inputTypeName;
+    private final List<MemberShape> inputMembersWithAttr;
+    private final String outputTypeName;
+    private final List<MemberShape> outputMembersWithAttr;
+
+    DocumentClientCommandGenerator(
+            TypeScriptSettings settings,
+            Model model,
+            OperationShape operation,
+            SymbolProvider symbolProvider,
+            TypeScriptWriter writer
+    ) {
+        this.model = model;
+        settings.getService(model);
+        this.operation = operation;
+        this.symbolProvider = symbolProvider;
+        this.writer = writer;
+
+        symbol = symbolProvider.toSymbol(operation);
+        operationIndex = OperationIndex.of(model);
+        inputTypeName = DocumentClientUtils.getModifiedName(
+            symbol.expectProperty("inputType", Symbol.class).getName()
+        );
+        inputMembersWithAttr = getStructureMembersWithAttr(operationIndex.getInput(operation));
+        outputTypeName = DocumentClientUtils.getModifiedName(
+            symbol.expectProperty("outputType", Symbol.class).getName()
+        );
+        outputMembersWithAttr = getStructureMembersWithAttr(operationIndex.getOutput(operation));
+    }
+
+    @Override
+    public void run() {
+        String serviceName = DocumentClientUtils.CLIENT_NAME;
+        String configType = DocumentClientUtils.CLIENT_CONFIG_NAME;
+
+        // Add required imports.
+        writer.addImport(configType, configType, "./" + serviceName);
+        writer.addImport("Command", "$Command", "@aws-sdk/smithy-client");
+
+        generateInputAndOutputTypes();
+
+        String name = DocumentClientUtils.getModifiedName(symbol.getName());
+        // writer.writeShapeDocs(operation);
+        writer.openBlock("export class $L extends $$Command<$L, $L, $L> {", "}",
+                name, inputTypeName, outputTypeName, configType, () -> {
+
+            // Section for adding custom command properties.
+            writer.write("// Start section: $L", COMMAND_PROPERTIES_SECTION);
+            writer.pushState(COMMAND_PROPERTIES_SECTION).popState();
+            writer.write("// End section: $L", COMMAND_PROPERTIES_SECTION);
+            writer.write("");
+
+            generateCommandConstructor();
+            writer.write("");
+            generateCommandMiddlewareResolver(configType);
+
+            // Hook for adding more methods to the command.
+            writer.write("// Start section: $L", COMMAND_BODY_EXTRA_SECTION)
+                    .pushState(COMMAND_BODY_EXTRA_SECTION)
+                    .popState()
+                    .write("// End section: $L", COMMAND_BODY_EXTRA_SECTION);
+        });
+    }
+
+    private void generateCommandConstructor() {
```

Comment:
This generator is producing constructor's that only call `super()`, which is redundant. Every single command is currently generated with a constructor that could be ommitted. I recommend re-working this so that a constructor is only generated when code is added to it; a `super()` only constructor should be removed.

Reply:
The constructor also sets class member `input` in addition to calling `super()` using [TypeScript parameter properties](https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties).

However, this is done by parent constructor too https://github.com/aws/aws-sdk-js-v3/blob/05ef1b4910a769b83c42c13da024e5d1567696e1/packages/smithy-client/src/client.ts#L24-L26 and is not required. I will rework to remove redundant constructor.

Reply:
**Update**: The constructor is required as I receive the following TypeScript errors:

On removing the constructor:
```console
src/commands/PutCommand.ts:34:14 - error TS2515: Non-abstract class 'PutCommand' does not implement inherited abstract member 'input' from class 'Command<PutCommandInput, PutCommandOutput, DynamoDBDocumentClientResolvedConfig, any, any>'.

34 export class PutCommand extends $Command<PutCommandInput, PutCommandOutput, DynamoDBDocumentClientResolvedConfig> {
```

On adding inherited abstract member `input`:
```console
src/commands/PutCommand.ts:35:12 - error TS2564: Property 'input' has no initializer and is not definitely assigned in the constructor.

35   readonly input: PutCommandInput;
```

Reply:
This does appear trickier than I originally thought. I'm trying to find a way to make typescript happy, but this might be a dead end